impeller/entity/geometry.cc

```diff
@@ -883,4 +892,135 @@ std::optional<Rect> RRectGeometry::GetCoverage(const Matrix& transform) const {
+      .index_buffer = host_buffer.Emplace(dummy_index.data(),
+                                          points_.size() * sizeof(uint32_t),
+                                          alignof(uint32_t)),
+      .index_count = points_.size(),
+      .index_type = IndexType::k32bit,
+  };
+
+  DeviceBufferDescriptor buffer_desc;
+  buffer_desc.size = total * sizeof(Point);
+  buffer_desc.storage_mode = StorageMode::kDevicePrivate;
+
+  auto buffer =
+      renderer.GetContext()->GetResourceAllocator()->CreateBuffer(buffer_desc);
+
+  auto index_buffer = host_buffer.Emplace(
+      dummy_index.data(), index_count * sizeof(uint32_t), alignof(uint32_t));
+
+  Command cmd;
+  cmd.label = "Points Geometry";
+
+  auto options = OptionsFromPass(pass);
+  options.blend_mode = BlendMode::kSource;
+  options.stencil_compare = CompareFunction::kAlways;
+  options.stencil_operation = StencilOperation::kKeep;
+  options.enable_rasterization = false;
+  cmd.pipeline = renderer.GetPointFieldGeometryPipeline(options);
+
+  VS::FrameInfo frame_info;
+  frame_info.radius = radius_;
+  frame_info.radian_start = round_ ? 0 : 0.785398;
+  frame_info.radian_step = k2Pi / divisions_per_circle;
+  frame_info.points_per_circle = divisions_per_circle * 3;
+  frame_info.divisions_per_circle = divisions_per_circle;
+
+  VS::BindFrameInfo(cmd, host_buffer.EmplaceUniform(frame_info));
+  VS::BindGeometryData(
+      cmd, {.buffer = buffer, .range = Range{0, total * sizeof(Point)}});
+
+  cmd.BindVertices(vtx_buffer);
+
+  if (!pass.AddCommand(std::move(cmd))) {
+    return {};
+  }
+
+  return {
+      .type = PrimitiveType::kTriangle,
+      .vertex_buffer = {.vertex_buffer = {.buffer = buffer,
+                                          .range =
+                                              Range{0, total * sizeof(Point)}},
+                        .index_buffer = index_buffer,
+                        .index_count = index_count,
+                        .index_type = IndexType::k32bit},
+      .transform = Matrix::MakeOrthographic(pass.GetRenderTargetSize()) *
+                   entity.GetTransformation(),
+      .prevent_overdraw = false,
+  };
+}
+
+GeometryResult PointFieldGeometry::GetPositionUVBuffer(
+    Rect texture_coverage,
+    Matrix effect_transform,
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass) {
+  FML_UNREACHABLE();
+}
+
+/// @brief Compute the exact storage size needed to store the resulting buffer.
+/// @return
+size_t PointFieldGeometry::ComputeCircleDivisions(Scalar scaled_radius,
+                                                  bool round) {
+  if (!round) {
+    return 4;
+  }
+  // note: this formula is completely arbitrary, we should find a reasonable
+  // curve based on experimental data.
+  if (scaled_radius < 4.0) {
+    return 8;
+  }
+  if (scaled_radius < 16) {
+    return 16;
+  }
+  if (scaled_radius < 32) {
+    return 32;
+  }
+  return 64;
```

Comment:
TODO

Reply:
See `SkNextPow2` and `SkNextLog2`.

Reply:
(Oh, I see, this is supposed to be modelling a curve not just always getting the next power of 2)